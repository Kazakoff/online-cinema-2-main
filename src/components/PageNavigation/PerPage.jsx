import React, { useRef, useEffect } from 'react';
import { DropDown } from '../DropDown';

export const PerPage = (props) => {

  const {
    limitArray,
    setLimit,
    limit,
    initialLimit
  } = props

  useEffect(() => {
    setLimit(initialLimit ? initialLimit : limitArray[0])
  }, [])

  return (
    <DropDown style={{ marginLeft: '7px' }} label={limit}>
      {
        limitArray.map(item => {
          return (
            <div
              key={item}
              className='d-flex justify-content-center w100'
              style={limit === item ? { backgroundColor: '#ccc', borderRadius: '100%' } : { cursor: 'pointer' }}
              onClick={() => setLimit(item)} >
              {item}
            </div>
          )
        })
      }
    </DropDown>
  )
}
