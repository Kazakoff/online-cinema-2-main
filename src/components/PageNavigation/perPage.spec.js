import React from 'react'
import { PerPage } from './PerPage'

const props = {
    limitArray: [1, 2, 3]
};
const setUp = (props) => shallow(<PerPage {...props} />);

describe('PerPage component', () => {
    it('should render PerPage component', () => {
        const component = setUp(props)
        expect(component.debug()).toMatchSnapshot()
    })
})