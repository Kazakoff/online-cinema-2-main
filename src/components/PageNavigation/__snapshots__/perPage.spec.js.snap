// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerPage component should render PerPage component 1`] = `
"<DropDown style={{...}} label={[undefined]}>
  <div className=\\"d-flex justify-content-center w100\\" style={{...}} onClick={[Function: onClick]}>
    1
  </div>
  <div className=\\"d-flex justify-content-center w100\\" style={{...}} onClick={[Function: onClick]}>
    2
  </div>
  <div className=\\"d-flex justify-content-center w100\\" style={{...}} onClick={[Function: onClick]}>
    3
  </div>
</DropDown>"
`;
